include/Animal.h

```diff
@@ -0,0 +1,110 @@
+#pragma once
+#include <iostream>
+#include <string>
+#include "Location.h"
+#include "Direction.h"
+
+/// <summary>
+/// Pure abstract Animal class
+/// </summary>
+class Animal {
+protected:
+	bool isMoving = true;
+	std::string name;
+	Location location;
+	Direction direction;
+	
+public:
+	/// <summary>
+	/// Random location - row between 0 and 19 and column between 0 and 39
+	/// </summary>
+	/// <returns>random location</returns>
+	static Location random();
+
+	/// <summary>
+	/// Default constructor
+	/// </summary>
+	Animal(): location() {};
+
```

Comment:
no use of virtual destructor 
check this
https://stackoverflow.com/questions/461203/when-to-use-virtual-destructors